### DISCUSSION TITLE
Assigning styles to `Select.ItemText` to allow for truncating of text

### DISCUSSION THREAD

**Discussion by dextermb** [radix-ui/primitives#2041] [Help]
Hey 👋🏻 

I'm looking at implementing Radix UI's Select but I've noticed that you cannot make the `Select.ItemText` truncate as it is within a `span` and we're unable to assign classes or styles to it.

```jsx
<RSelect.Root>
  <RSelect.Trigger asChild>
    <div
      className={cn(
        'flex flex-shrink-0 justify-between items-center w-64 gap-2 px-3 py-2 text-sm border rounded-lg',
        className
      )}
    >
      <RSelect.Value
        className='truncate' {/* <-- This class is not applied */}
        placeholder={placeholder}
      />
      <RSelect.Icon asChild>
        <Icon src={<ChevronUpDownIcon />} />
      </RSelect.Icon>
    </div>
  </RSelect.Trigger>

  <RSelect.Portal>
    <RSelect.Content className='p-1 space-y-px text-sm bg-white rounded-md shadow min-w-32'>
      <RSelect.ScrollUpButton asChild>
        <Icon src={<ChevronUpIcon />} />
      </RSelect.ScrollUpButton>
      <RSelect.Viewport>
        {options.map(option => (
          <RSelect.Item
            key={option.text}
            value={option.value}
            className={cn(
              'flex justify-between items-center gap-1 px-2 py-1.5 rounded-md cursor-pointer',
              'focus:outline-none active:outline-none',
              'data-[highlighted]:bg-zinc-50 hover:!bg-zinc-100',
              option.theme === 'danger' && 'text-red-500'
            )}
          >
            <RSelect.ItemText
              className='truncate' {/* <-- This class is not applied */}
              asChild
            >
              <div className='flex items-center justify-start min-w-0 gap-1'>
                {option.icon
                  ? <Icon src={option.icon} />
                  : <span className='w-4 h-4 shrink-0' />}
                <span className='truncate'>{option.text}</span>
              </div>
            </RSelect.ItemText>
            <RSelect.ItemIndicator asChild>
              <Icon src={<CheckIcon />} />
            </RSelect.ItemIndicator>
          </RSelect.Item>
        ))}
      </RSelect.Viewport>
      <RSelect.ScrollDownButton asChild>
        <Icon src={<ChevronDownIcon />} />
      </RSelect.ScrollDownButton>
    </RSelect.Content>
  </RSelect.Portal>
</RSelect.Root>
```

| Without truncate | With truncate |
|------------------|---------------|
| <img width="1531" alt="Screenshot 2023-03-23 at 16 33 37" src="https://user-images.githubusercontent.com/6137789/227272268-9d0087a2-23f0-4bf2-91c2-7bb7b5d66108.png"> | <img width="1531" alt="Screenshot 2023-03-23 at 16 33 28" src="https://user-images.githubusercontent.com/6137789/227272355-d937d6b1-1f58-4595-860c-9076ab66708b.png"> |

Within my component I accept an array of `options`, an attribute on an option is `text`, this text can be longer than the input (especially when stubbing the input short).

As I'm using Tailwind, and a complex item, I need to assign the `truncate` class to the surrounding `span` element created by `RSelect.ItemText` but classes are not passed along to the span.

Why the docs say it does not encourage styling the `ItemText` in these cases we should be able to.

---

Any ideas for the best work around?

Edit 1: Continued investigation seems like it's the `RSelect.Value` that requires the ability to accept classes or styles – not `Select.ItemText`

**Comment by joaom00** [comment]
You can use arbitrary selector with tailwind `[&>span:first-child]:truncate` on `Select.Trigger`

**Reply by dextermb** [reply]
> You can use arbitrary selector with tailwind `[&>span:first-child]:truncate` on `Select.Trigger`

I'll give this a go. Thanks 🙏🏻

**Comment by benoitgrelard** [comment]
We don't allow styling those elements as they are intended to only contain the text so it can be aligned properly between the value in the trigger and the item's text.

You can however simply wrap it with an extra element where you apply whichever styles you want.
